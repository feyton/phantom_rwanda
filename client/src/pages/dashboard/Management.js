import CheckRole from '@utils/CheckRoles.js';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Tab, TabList, TabPanel, Tabs } from 'react-tabs';
import { setActiveTab } from '../../redux/reducers/managementReducer.js';
import Buses from '../management/Buses.js';
import Companies from '../management/Companies.js';
import Drivers from '../management/Drivers.js';
import Operators from '../management/Operators.js';
import Routes from '../management/Routes.js';

const Management = () => {
  const { tabIndex } = useSelector((state) => state?.management);
  const dispatch = useDispatch();
  const allTabs = [
    { name: 'Drivers', roles: ['operator', 'admin', 'driver'] },
    { name: 'Buses', roles: ['operator', 'admin'] },
    { name: 'Routes', roles: ['admin', 'operator'] },
    { name: 'Companies', roles: ['admin'] },
    { name: 'Operators', roles: ['admin'] }
  ];
  const handleSelect = (index) => {
    dispatch(setActiveTab(index));
  };
  return (
    <main className="overflow-x-auto">
      <Tabs
        selectedIndex={tabIndex}
        onSelect={(index) => handleSelect(index)}
        selectedTabClassName="border-b-2 border-primary h-full mb-0"
      >
        <TabList className="flex border-b-2 mb-5 w-full justify-around shadow-main">
          {allTabs.map((tab) => (
            <CheckRole
              key={tab.name}
              children={
                <Tab
                  className="mr-16 font-bold py-2 w-full px-3 text-center cursor-pointer outline-none focus:outline-none font-raleway"
                  key={tab.name}
                >
                  {tab.name}
                </Tab>
              }
              role={tab.roles}
            />
          ))}
        </TabList>
        <TabPanel>
          <CheckRole
            children={<Drivers />}
            role={['operator', 'admin', 'driver']}
          />
        </TabPanel>

        <TabPanel>
          <CheckRole
            children={<Buses />}
            type="page"
            role={['operator', 'admin']}
          />
        </TabPanel>
        <TabPanel>
          <CheckRole children={<Routes />} role={['admin', 'operator']} />
        </TabPanel>
        <TabPanel>
          <CheckRole children={<Companies />} role={['admin']} />
        </TabPanel>
        <TabPanel>
          {' '}
          <CheckRole children={<Operators />} role={['admin']} />{' '}
        </TabPanel>
      </Tabs>
    </main>
  );
};

export default Management;
