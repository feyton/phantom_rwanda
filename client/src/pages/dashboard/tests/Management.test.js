import { render } from '@testing-library/react';
import ProviderWrapper from '@utils/TestUtil';
import React from 'react';
import Management from '../Management.js';
import Roles from '../RolesPage.js';

describe('Management & Roles', () => {
  it('Render the management', () => {
    const elem = render(<ProviderWrapper children={<Management />} />);
    expect(elem).toMatchSnapshot();
  });
  it('Render the role page', () => {
    const elem = render(<ProviderWrapper children={<Roles />} />);
    expect(elem).toMatchSnapshot();
  });
});
