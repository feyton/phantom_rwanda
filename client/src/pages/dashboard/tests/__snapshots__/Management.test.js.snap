// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Management & Roles Render the management 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="overflow-x-auto"
      >
        <div
          class="react-tabs"
          data-rttabs="true"
        >
          <ul
            class="flex border-b-2 mb-5 w-full justify-around shadow-main"
            role="tablist"
          >
            
            
            
            
            
          </ul>
          <div
            aria-labelledby="react-tabs-0"
            class="react-tabs__tab-panel react-tabs__tab-panel--selected"
            id="react-tabs-1"
            role="tabpanel"
          >
            
          </div>
          <div
            aria-labelledby="react-tabs-2"
            class="react-tabs__tab-panel"
            id="react-tabs-3"
            role="tabpanel"
          />
          <div
            aria-labelledby="react-tabs-4"
            class="react-tabs__tab-panel"
            id="react-tabs-5"
            role="tabpanel"
          />
          <div
            aria-labelledby="react-tabs-6"
            class="react-tabs__tab-panel"
            id="react-tabs-7"
            role="tabpanel"
          />
          <div
            aria-labelledby="react-tabs-8"
            class="react-tabs__tab-panel"
            id="react-tabs-9"
            role="tabpanel"
          />
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="overflow-x-auto"
    >
      <div
        class="react-tabs"
        data-rttabs="true"
      >
        <ul
          class="flex border-b-2 mb-5 w-full justify-around shadow-main"
          role="tablist"
        >
          
          
          
          
          
        </ul>
        <div
          aria-labelledby="react-tabs-0"
          class="react-tabs__tab-panel react-tabs__tab-panel--selected"
          id="react-tabs-1"
          role="tabpanel"
        >
          
        </div>
        <div
          aria-labelledby="react-tabs-2"
          class="react-tabs__tab-panel"
          id="react-tabs-3"
          role="tabpanel"
        />
        <div
          aria-labelledby="react-tabs-4"
          class="react-tabs__tab-panel"
          id="react-tabs-5"
          role="tabpanel"
        />
        <div
          aria-labelledby="react-tabs-6"
          class="react-tabs__tab-panel"
          id="react-tabs-7"
          role="tabpanel"
        />
        <div
          aria-labelledby="react-tabs-8"
          class="react-tabs__tab-panel"
          id="react-tabs-9"
          role="tabpanel"
        />
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Management & Roles Render the role page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full h-full font-raleway px-10 pt-3"
      >
        <h1
          class="text-2xl font-bold text-center justify-center font-rale"
        >
          Manage roles and permissions
        </h1>
        <div
          class="mt-6 p-5 rounded-md shadow-main"
        >
          <div
            class="w-full border rounded-sm p-2 "
          >
            <div>
              <ul
                class="grid grid-cols-[30%_60%] gap-1 border-b-2 font-raleway  font-bold py-1 px-2"
              >
                <li>
                  Role
                </li>
                <ul
                  class="grid grid-cols-2"
                >
                  <li>
                    Resources
                  </li>
                  <li>
                    Permissions
                  </li>
                </ul>
              </ul>
            </div>
            <div
              class="border-b-2"
            >
              <ul
                class="font-raleway capitalize grid grid-cols-[30%_60%] gap-1 border-b-2 py-1"
              >
                <li
                  class="font-bold  text-primary"
                >
                  <span
                    class="cursor-pointer"
                  >
                    admin
                  </span>
                </li>
                <li>
                  <ul>
                    <div
                      class="grid grid-cols-2 py-2 justify-center center border-b "
                    >
                      <li
                        class="text-black "
                      >
                        driver
                      </li>
                      <li
                        class="py-1 flex flex-wrap lg:inline-block"
                      >
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          get
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-create bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          create
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-update bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          update
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-delete bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          delete
                        </span>
                      </li>
                    </div>
                    <div
                      class="grid grid-cols-2 py-2 justify-center center border-b "
                    >
                      <li
                        class="text-black "
                      >
                        routes
                      </li>
                      <li
                        class="py-1 flex flex-wrap lg:inline-block"
                      >
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          get
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-create bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          create
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-update bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          update
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-delete bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          delete
                        </span>
                      </li>
                    </div>
                    <div
                      class="grid grid-cols-2 py-2 justify-center center border-b "
                    >
                      <li
                        class="text-black "
                      >
                        operator
                      </li>
                      <li
                        class="py-1 flex flex-wrap lg:inline-block"
                      >
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          get
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-create bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          create
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-update bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          update
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-delete bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          delete
                        </span>
                      </li>
                    </div>
                    <div
                      class="grid grid-cols-2 py-2 justify-center center border-b "
                    >
                      <li
                        class="text-black "
                      >
                        company
                      </li>
                      <li
                        class="py-1 flex flex-wrap lg:inline-block"
                      >
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          get
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-create bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          create
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-update bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          update
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-delete bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          delete
                        </span>
                      </li>
                    </div>
                  </ul>
                </li>
                <li />
              </ul>
              <ul
                class="font-raleway capitalize grid grid-cols-[30%_60%] gap-1 border-b-2 py-1"
              >
                <li
                  class="font-bold  text-primary"
                >
                  <span
                    class="cursor-pointer"
                  >
                    operator
                  </span>
                </li>
                <li>
                  <ul>
                    <div
                      class="grid grid-cols-2 py-2 justify-center center border-b "
                    >
                      <li
                        class="text-black "
                      >
                        driver
                      </li>
                      <li
                        class="py-1 flex flex-wrap lg:inline-block"
                      >
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          get
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-create bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          create
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-update bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          update
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-delete bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          delete
                        </span>
                      </li>
                    </div>
                    <div
                      class="grid grid-cols-2 py-2 justify-center center border-b "
                    >
                      <li
                        class="text-black "
                      >
                        routes
                      </li>
                      <li
                        class="py-1 flex flex-wrap lg:inline-block"
                      >
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          get
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-create bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          create
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-update bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          update
                        </span>
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-delete bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          delete
                        </span>
                      </li>
                    </div>
                  </ul>
                </li>
                <li />
              </ul>
              <ul
                class="font-raleway capitalize grid grid-cols-[30%_60%] gap-1 border-b-2 py-1"
              >
                <li
                  class="font-bold  text-primary"
                >
                  <span
                    class="cursor-pointer"
                  >
                    driver
                  </span>
                </li>
                <li>
                  <ul>
                    <div
                      class="grid grid-cols-2 py-2 justify-center center border-b "
                    >
                      <li
                        class="text-black "
                      >
                        driver
                      </li>
                      <li
                        class="py-1 flex flex-wrap lg:inline-block"
                      >
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          get
                        </span>
                      </li>
                    </div>
                    <div
                      class="grid grid-cols-2 py-2 justify-center center border-b "
                    >
                      <li
                        class="text-black "
                      >
                        routes
                      </li>
                      <li
                        class="py-1 flex flex-wrap lg:inline-block"
                      >
                        <span
                          class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                        >
                          get
                        </span>
                      </li>
                    </div>
                  </ul>
                </li>
                <li />
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full h-full font-raleway px-10 pt-3"
    >
      <h1
        class="text-2xl font-bold text-center justify-center font-rale"
      >
        Manage roles and permissions
      </h1>
      <div
        class="mt-6 p-5 rounded-md shadow-main"
      >
        <div
          class="w-full border rounded-sm p-2 "
        >
          <div>
            <ul
              class="grid grid-cols-[30%_60%] gap-1 border-b-2 font-raleway  font-bold py-1 px-2"
            >
              <li>
                Role
              </li>
              <ul
                class="grid grid-cols-2"
              >
                <li>
                  Resources
                </li>
                <li>
                  Permissions
                </li>
              </ul>
            </ul>
          </div>
          <div
            class="border-b-2"
          >
            <ul
              class="font-raleway capitalize grid grid-cols-[30%_60%] gap-1 border-b-2 py-1"
            >
              <li
                class="font-bold  text-primary"
              >
                <span
                  class="cursor-pointer"
                >
                  admin
                </span>
              </li>
              <li>
                <ul>
                  <div
                    class="grid grid-cols-2 py-2 justify-center center border-b "
                  >
                    <li
                      class="text-black "
                    >
                      driver
                    </li>
                    <li
                      class="py-1 flex flex-wrap lg:inline-block"
                    >
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        get
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-create bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        create
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-update bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        update
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-delete bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        delete
                      </span>
                    </li>
                  </div>
                  <div
                    class="grid grid-cols-2 py-2 justify-center center border-b "
                  >
                    <li
                      class="text-black "
                    >
                      routes
                    </li>
                    <li
                      class="py-1 flex flex-wrap lg:inline-block"
                    >
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        get
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-create bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        create
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-update bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        update
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-delete bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        delete
                      </span>
                    </li>
                  </div>
                  <div
                    class="grid grid-cols-2 py-2 justify-center center border-b "
                  >
                    <li
                      class="text-black "
                    >
                      operator
                    </li>
                    <li
                      class="py-1 flex flex-wrap lg:inline-block"
                    >
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        get
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-create bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        create
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-update bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        update
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-delete bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        delete
                      </span>
                    </li>
                  </div>
                  <div
                    class="grid grid-cols-2 py-2 justify-center center border-b "
                  >
                    <li
                      class="text-black "
                    >
                      company
                    </li>
                    <li
                      class="py-1 flex flex-wrap lg:inline-block"
                    >
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        get
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-create bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        create
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-update bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        update
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-delete bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        delete
                      </span>
                    </li>
                  </div>
                </ul>
              </li>
              <li />
            </ul>
            <ul
              class="font-raleway capitalize grid grid-cols-[30%_60%] gap-1 border-b-2 py-1"
            >
              <li
                class="font-bold  text-primary"
              >
                <span
                  class="cursor-pointer"
                >
                  operator
                </span>
              </li>
              <li>
                <ul>
                  <div
                    class="grid grid-cols-2 py-2 justify-center center border-b "
                  >
                    <li
                      class="text-black "
                    >
                      driver
                    </li>
                    <li
                      class="py-1 flex flex-wrap lg:inline-block"
                    >
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        get
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-create bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        create
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-update bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        update
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-delete bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        delete
                      </span>
                    </li>
                  </div>
                  <div
                    class="grid grid-cols-2 py-2 justify-center center border-b "
                  >
                    <li
                      class="text-black "
                    >
                      routes
                    </li>
                    <li
                      class="py-1 flex flex-wrap lg:inline-block"
                    >
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        get
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-create bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        create
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-update bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        update
                      </span>
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-delete bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        delete
                      </span>
                    </li>
                  </div>
                </ul>
              </li>
              <li />
            </ul>
            <ul
              class="font-raleway capitalize grid grid-cols-[30%_60%] gap-1 border-b-2 py-1"
            >
              <li
                class="font-bold  text-primary"
              >
                <span
                  class="cursor-pointer"
                >
                  driver
                </span>
              </li>
              <li>
                <ul>
                  <div
                    class="grid grid-cols-2 py-2 justify-center center border-b "
                  >
                    <li
                      class="text-black "
                    >
                      driver
                    </li>
                    <li
                      class="py-1 flex flex-wrap lg:inline-block"
                    >
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        get
                      </span>
                    </li>
                  </div>
                  <div
                    class="grid grid-cols-2 py-2 justify-center center border-b "
                  >
                    <li
                      class="text-black "
                    >
                      routes
                    </li>
                    <li
                      class="py-1 flex flex-wrap lg:inline-block"
                    >
                      <span
                        class="mx-2 my-1 lg:my-0 uppercase bg-get bg-opacity-75 cursor-pointer text-white py-1 px-2 rounded-sm font-bold"
                      >
                        get
                      </span>
                    </li>
                  </div>
                </ul>
              </li>
              <li />
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
